enhancements/testing/ephemeral-agent-installer-e2e-tests.md

```diff
@@ -0,0 +1,373 @@
+authors:
+  - "@lranjbar"
+reviewers:
+  - "@rwsu"
+  - "@zaneb"
+  - "@pawanpinjarkar"
+approvers:
+  - "@dhellmann"
+  - "@celebdor"
+api-approvers: # in case of new or modified APIs or API extensions (CRD, aggregated api servers, webhooks, finalizers)
+  - N/A
+creation-date: 2022-02-10
+last-updated: 2022-03-09
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/AGENT-20
+---
+
+# Ephemeral Agent Installer e2e Tests
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Create a new e2e testing framework for the new agent based installer project. 
+Currently, for this project there would be potentially two existing e2e frameworks 
+that could be used, [dev-scripts](https://github.com/openshift-metal3/dev-scripts) 
+and [assisted-test-infra](https://github.com/openshift/assisted-test-infra).
+
+After discussing this proposal with the Metal Platform team we will be improving 
+dev-scripts components that overlap in the e2e flow desired for the ephemeral agent 
+installer project. 
+
+## Motivation
+
+The new agent based installer project will need an e2e testing framework for validation.
+We should look at the existing frameworks and understand what re-purposing them will 
+entail for this new project.
+
+This enhancement looks at the existing e2e frameworks [dev-scripts](https://github.com/openshift-metal3/dev-scripts) 
+and [assisted-test-infra](https://github.com/openshift/assisted-test-infra) since this 
+is the previous work in this space. The motivation is to expand on what we have learned 
+from those frameworks try to come up with a solution that we can reuse for new installers.
+
+### Goals
+
+* This proposal will focus using this new e2e framework in the new ephemeral agent based 
+installer image project.
+
+### Non-Goals
+
+* Re-factoring dev-scripts and assisted-test-infra for reasons not related to the new 
+ephemeral agent based installer image project.
+* There should be some overlap in the pieces that are refactored for use in the new project.
+However it is still a non-goal for outside installer to use this new framework in their e2e 
+tests for the upcoming release. We will plan to create a point of convergence after the
+ephemeral agent based installer project is out of prototyping and MVP phase.
+
+## Proposal
+
+Create an e2e testing framework for the ephemeral agent based installer image in the 
+[dev-scripts](https://github.com/openshift-metal3/dev-scripts) repository and refactor 
+components as needed for the new project.
```

Comment:
I would suggest adding more details about the framework being proposed here. I had to scan down the document and saw there are more specific information in the Design Details. Being able to see what the framework being proposed looks like here, will aid in helping understand how they impact the User Stories discussed in the next section.

Reply:
I took this into account and put in more information about file names and things I've identified in the file structure part. I was hesitant before because those things may change when they get into code review in dev-scripts. I think though this will help clarify things a lot so I added it in.

Reply:
An update to this comment. We decided to go with a more piece by piece flow for some of these things. This is now described in the design details section